eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/configuration/batch/io/RSMergeAndSplitCandidatesReaderConfiguration.java

```diff
@@ -48,53 +49,53 @@ public class RSMergeAndSplitCandidatesReaderConfiguration {
+    public static Criteria getSplitCandidatesCriteria(String assemblyAccession) {
+        return where(DBCollection.ID_FIELD_NAME).regex("^RSSC_" + assemblyAccession + "_.*");
     }
 
-    public static Query getMergeCandidatesQuery(String assemblyAccession) {
-        return Query.query(where(ASSEMBLY_FIELD).is(assemblyAccession))
-                    .addCriteria(where(EVENT_TYPE_FIELD).is(MERGE_CANDIDATES_EVENT_TYPE.toString()));
+    public static Criteria getMergeCandidatesCriteria(String assemblyAccession) {
+        return where(DBCollection.ID_FIELD_NAME).regex("^RSMC_" + assemblyAccession + "_.*");
     }
 
     @Bean(RS_SPLIT_CANDIDATES_READER)
-    public ItemReader<SubmittedVariantOperationEntity> rsSplitCandidatesReader(MongoTemplate mongoTemplate,
-                                                                               InputParameters parameters) {
-        MongoItemReader<SubmittedVariantOperationEntity> mongoItemReader = new MongoItemReader<>();
+    public MongoDbCursorItemReader<SubmittedVariantOperationEntity> rsSplitCandidatesReader(MongoTemplate mongoTemplate,
+                                                                                            InputParameters parameters)
+    {
+        MongoDbCursorItemReader<SubmittedVariantOperationEntity> mongoItemReader = new MongoDbCursorItemReader<>();
         mongoItemReader.setTemplate(mongoTemplate);
         mongoItemReader.setTargetType(SubmittedVariantOperationEntity.class);
         mongoItemReader.setCollection(SUBMITTED_VARIANT_OPERATIONS_COLLECTION);
 
         //See https://docs.google.com/spreadsheets/d/1KQLVCUy-vqXKgkCDt2czX6kuMfsjfCc9uBsS19MZ6dY/#rangeid=1213746442
-        Query query = getSplitCandidatesQuery(parameters.getAssemblyAccession());
+        Meta meta = new Meta();
+        meta.addFlag(Meta.CursorOption.NO_TIMEOUT);
+        Query query = query(getSplitCandidatesCriteria(parameters.getAssemblyAccession()));
+        query.setMeta(meta);
         mongoItemReader.setQuery(query);
-        mongoItemReader.setPageSize(parameters.getChunkSize());
-        mongoItemReader.setSort(Collections.singletonMap(SORT_FIELD, Sort.Direction.ASC));
         return mongoItemReader;
     }
 
     @Bean(RS_MERGE_CANDIDATES_READER)
-    public ItemReader<SubmittedVariantOperationEntity> rsMergeCandidatesReader(MongoTemplate mongoTemplate,
-                                                                               InputParameters parameters) {
-        MongoItemReader<SubmittedVariantOperationEntity> mongoItemReader = new MongoItemReader<>();
+    public MongoDbCursorItemReader<SubmittedVariantOperationEntity> rsMergeCandidatesReader(MongoTemplate mongoTemplate,
+                                                                                            InputParameters parameters)
+    {
+        MongoDbCursorItemReader<SubmittedVariantOperationEntity> mongoItemReader = new MongoDbCursorItemReader<>();
```

Comment:
This might take care of performance issues due to paging as well as consistency issue such as EVA-2867 (if we are lucky). See [here](https://stackoverflow.com/a/45890047)